src/App.tsx

```diff
@@ -1,24 +1,227 @@
+        const loadedTodos = await getTodos(USER_ID);
+
+        setTodos(loadedTodos);
+        setError('');
+      } catch {
+        setError('Unable to add a todo');
+        setTimeout(() => setError(''), 3000);
+      }
+    }, [getTodos],
+  );
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const vidibleTodos = useMemo(() => {
+    return filteredTodos(todos, status);
+  }, [todos, status]);
+
+  const addTodo = useCallback(
+    async (title: string) => {
+      setLoading(true);
+      if (!title) {
+        setError('Title can t be empty');
+        setTimeout(() => setError(''), 3000);
+      }
+
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+
+      try {
+        await addTodoOnServer(newTodo);
+        setTempTodo(newTodo);
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        addLoadingTodo(newTodo.id);
+      } catch {
+        setError('Unable to add todo!');
+        setTimeout(() => setError(''), 3000);
+      } finally {
+        setLoading(false);
+        removeLoadingTodo(newTodo.id);
+        setTempTodo(null);
+      }
+    }, [addLoadingTodo, removeLoadingTodo],
+  );
+
+  const deleteTodo = useCallback(
+    async (todoId: number) => {
+      setLoading(true);
+      addLoadingTodo(todoId);
+
+      try {
+        await deleteTodoFromServer(todoId);
+        setTodos(currentTodos => (
+          currentTodos.filter(todo => todo.id !== todoId)
+        ));
+      } catch {
+        setError('Unable to delete a todo!');
+        setTimeout(() => setError(''), 3000);
+      } finally {
+        setLoading(false);
+        removeLoadingTodo(todoId);
+      }
+    }, [addLoadingTodo, removeLoadingTodo],
+  );
+
+  const deleteComplitedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(({ id }) => {
+      deleteTodo(id);
+    });
+
+    setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+  };
+
+  const updateTodo = useCallback(
+    async (id: number, todoData: Partial<Todo>) => {
+      addLoadingTodo(id);
+      try {
+        const updatedTodo = await updateTodoOnServer(id, todoData);
+
+        setTodos(currentTodos => currentTodos.map(
+          todo => (todo.id === updatedTodo.id
+            ? updatedTodo
+            : todo),
+        ));
+      } catch {
+        setError('Unable to update a todo!');
+        setTimeout(() => setError(''), 3000);
+      } finally {
+        removeLoadingTodo(id);
+      }
+    }, [addLoadingTodo, removeLoadingTodo],
+  );
+
+  const updateTodoAll = () => {
+    const completedAllTodos = todos.every(todo => todo.completed === true);
+
+    if (completedAllTodos) {
+      todos.forEach(todo => {
+        updateTodo(todo.id, { completed: false });
+      });
+    } else {
+      todos.forEach(todo => {
+        updateTodo(todo.id, { completed: true });
+      });
+    }
```

Comment:
there's an easy way to write this with ternary operator

Reply:
Actually, you can just do it like this:
```js
todos.forEach(todo => {
  updateTodo(todo.id, { completed: !completedAllTodos });
});
```